ui/app/src/components/BluePrintForm.tsx

```diff
@@ -0,0 +1,153 @@
+import FormBuilder from "./FormBuilder";
+
+const useStyles = makeStyles(() => ({
+  form: {
+    maxWidth: '600px',
+    margin: 'auto',
+    paddingBottom: '20px'
+  }
+}));
+
+interface BluePrintForm {
+  inputs: SiteState,
+  setInputs(state: SiteState): any,
+  onSubmit(event: any): any,
+  swipeableViews: any,
+  blueprint: Blueprint,
+  onCheckNameExist(event: any): any
+}
+
+const messages = defineMessages({
+  siteId: {
+    id: 'createSiteDialog.siteId',
+    defaultMessage: 'Side ID'
+  },
+  description: {
+    id: 'createSiteDialog.description',
+    defaultMessage: 'Description'
+  },
+  siteFormat: {
+    id: 'createSiteDialog.siteFormat',
+    defaultMessage: 'Max length: 50 characters, consisting of: lowercase letters, numbers, dash (-) and underscore (_)'
+  },
+  nameExist: {
+    id: 'createSiteDialog.nameExist',
+    defaultMessage: 'The name already exist'
+  },
+  pushSiteToRemote: {
+    id: 'createSiteDialog.pushSiteToRemote',
+    defaultMessage: 'Push the site to a remote Git repository after creation'
+  },
+  descriptionMaxLength: {
+    id: 'createSiteDialog.descriptionMaxLength',
+    defaultMessage: 'Max length: 4000 characters'
+  },
+});
+
+function BluePrintForm(props: BluePrintForm) {
+  const classes = useStyles({});
+  const {inputs, setInputs, onSubmit, swipeableViews, blueprint, onCheckNameExist} = props;
+  const { formatMessage } = useIntl();
+
+  useEffect(
+    () => {
+      if (swipeableViews) {
+        const timer = setTimeout(() => swipeableViews.current.updateHeight(), 300);
+        return () => clearTimeout(timer);
+      }
+    },
+    // eslint-disable-next-line
+    [inputs.pushSite, inputs.repoAuthentication],
+  );
+
+  const handleInputChange = (e: any, type?:string) => {
+    e.persist();
+    if (e.target.type === 'checkbox') {
+      setInputs({...inputs, [e.target.name]: e.target.checked});
+    } else if (e.target.name === 'siteId') {
+      setInputs({...inputs, [e.target.name]: e.target.value.replace(/\s+/g, "").toLowerCase()});
+    } else if (type === 'blueprintFields') {
+      let parameters = {...inputs.blueprintFields, [e.target.name]: e.target.value };
+      setInputs({...inputs, blueprintFields: parameters});
+    } else {
+      setInputs({...inputs, [e.target.name]: e.target.value});
+    }
+  };
+
+  return (
+    <form className={classes.form} onSubmit={e => onSubmit(e)}>
+      <Grid container spacing={3}>
+        <Grid item xs={12}>
+          <TextField
+            id="siteId"
+            name="siteId"
+            label={formatMessage(messages.siteId)}
+            required
+            fullWidth
+            onBlur={onCheckNameExist}
+            onChange={(event) => handleInputChange(event)}
+            value={inputs.siteId}
+            error={((inputs.submitted && !inputs.siteId) || inputs.siteIdExist)}
+            helperText={!inputs.siteIdExist ? formatMessage(messages.siteFormat) : formatMessage(messages.nameExist)}
```

Comment:
Use message formats (params) to supply the max characters. That way you can use the same const here for validation and pass it on to the localized message.